src/common/utils.py

```diff
@@ -421,3 +431,30 @@ def grouper(iterable, n):
         if not chunk:
             return
         yield chunk
+
+
+def load_additional_instance_types_data(config, section):
+    """Load instance types data from configuration, if set; an empty dict is returned otherwise."""
+    instance_types_data = {}
+    if config.has_option(section, "instance_types_data"):
+        instance_types_data_str = config.get(section, "instance_types_data")
```

Comment:
optional: you could directly do `config.get(section, "instance_type_data", "{}").strip()` and avoid a bunch of ifs